### PR TITLE
Addressed PR comments.

### PR DESCRIPTION
**Pull request by shashank-parulekar**
Signed-off-by: Shashank Parulekar <shashank.parulekar@seagate.com>

# Problem Statement
- Addressed PR comments in the Btree and related code.

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Review Checklist 
- [ ] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

### CODE REVIEW

File: btree/btree.h [hunk 1]
```diff
@@ -205,6 +205,37 @@ enum m0_btree_purge_user{
 	M0_PU_EXTERNAL,
 };
 
+
```
Comment (ydb242) [inline btree/btree.h:208]: Extra line. Can be removed.

File: btree/btree.h [hunk 2]
```diff
@@ -205,6 +205,37 @@ enum m0_btree_purge_user{
...
+		M0_CASSERT(M0_HAS_TYPE((p_src), struct m0_btree_rec *));       \
+		m0_bufvec_copy(&__tgt_rec->r_key.k_data,                       \
+			       &__src_rec ->r_key.k_data,                      \
+			       m0_vec_count(&__src_rec ->r_key.k_data.ov_vec));\
+		m0_bufvec_copy(&__tgt_rec->r_val,                              \
+			       &__src_rec->r_val,                              \
+			       m0_vec_count(&__src_rec ->r_val.ov_vec));       \
+	})
+
+
```
Comment (ydb242) [inline btree/btree.h:238]: Extra line. Can be removed.

File: btree/btree.h (hunk 1 above)
Comment (shashank-parulekar) [reply btree/btree.h]: done

File: btree/btree.h (hunk 2 above)
Comment (shashank-parulekar) [reply btree/btree.h]: done

**Review by ydb242 — APPROVED** [review]

**Review by kanchan-chaudhari — APPROVED** [review]

**Review by upendrapatwardhan — APPROVED** [review]

**Pull request merged**